src/Containers/Article/Edit/editArticle.jsx

```diff
@@ -0,0 +1,294 @@
+import React, { Component } from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+import { ImageSideButton, Block, addNewBlock, createEditorState, Editor } from 'medium-draft';
+import { convertToRaw } from 'draft-js';
+import mediumDraftImporter from 'medium-draft/lib/importer';
+import axios from 'axios';
+import LoadingBar from 'react-top-loading-bar';
+import { ToastContainer } from 'react-toastify';
+import mediumDraftExporter from 'medium-draft/lib/exporter';
+import { faFeather } from '@fortawesome/free-solid-svg-icons';
+import { getArticle, updateArticle } from '../../../actions/article';
+import getUser from '../../../actions/userInfo';
+import Sidebar from '../../../Components/SideBar/SideBar';
+import Navbar from '../../../Components/NavBar/NavBar';
+import Button from '../../../Components/Common/Button/Button';
+import 'medium-draft/lib/index.css';
+import './editArticle.scss';
+import 'react-toastify/dist/ReactToastify.css';
+
+let images = [];
+let spinner = false;
+
+/**
+ *
+ * @param {object} editorState
+ * @returns {method} render
+ */
+export class CustomImageSideButton extends ImageSideButton {
+  /*
+  We will only check for first file and also whether
+  it is an image or not.
+  */
+  onChange = e => {
+    spinner = true;
+    const file = e.target.files[0];
+    if (file.type.indexOf('image/') === 0) {
+      // This is a post request to server endpoint with image as `image`
+      const formData = new FormData();
+      formData.append('file', file, file.name);
+      formData.append('upload_preset', process.env.REACT_APP_CLOUDINARY_PRESET);
+      formData.append('api_key', process.env.REACT_APP_CLOUDINARY_API_KEY);
+      axios
+        .post(process.env.REACT_APP_CLOUDINARY_URL, formData, {
+          headers: { 'X-Requested-With': 'XMLHttpRequest' },
+        })
+        .then(response => {
+          const { data } = response;
+          spinner = false;
+          images = [...images, data.url];
+          if (data.url) {
+            this.props.setEditorState(
+              addNewBlock(this.props.getEditorState(), Block.IMAGE, {
+                src: data.url,
+              }),
+            );
+          }
+        });
+    }
+    this.props.close();
+  };
+}
+
+/**
+ *
+ * @param {e} e
+ * @returns {method} render
+ */
+export class EditArticle extends Component {
+  state = {
+    title: '',
+    editorState: createEditorState(),
+    loadingBarProgress: 0,
+    contentSetted: false,
+  };
+
+  sideButtons = [
+    {
+      title: 'Image',
+      component: CustomImageSideButton,
+    },
+  ];
+
+  /**
+   * @return {object}
+   * s {jsx} react fragment
+   */
+  componentDidMount() {
+    this.setState({ publish: false });
+    this.refs.editor.focus();
+    const {
+      onGetUserInfo,
+      currentUser,
+      onGetArticle,
+      match: {
+        params: { slug },
+      },
+    } = this.props;
+    onGetUserInfo(currentUser.user.username);
```

Comment:
@dusmel Consider having deep destructuring. `currentUser: {user: {username}}`

Reply:
You can check my component will unmount to see that I use destructuring properly, but sometimes I use it with caution because I value most code readability than complex syntax and I feel destructuring on more than one level makes my code less readable in some cases.